validator/src/validator.rs

```diff
@@ -232,6 +237,50 @@ where
         self.init_block_producer(head_hash);
     }
 
+    fn push_block_timestamp(&mut self, timestamp: u64) {
+        // Only push the timestamps once caught up with the chain
+        if !self.is_synced() {
+            return;
+        }
+
+        let amount_of_timestamps = self.recent_block_timestamps.len();
+        if amount_of_timestamps >= 30 {
+            // The VeqDeque should not grow unbounded
+            assert!(amount_of_timestamps < 31);
+
+            self.recent_block_timestamps.pop_front();
+        }
+
+        self.recent_block_timestamps.push_back(timestamp);
+    }
+
+    fn compute_producer_timeout(&self) -> u64 {
+        let amount_of_timestamps = self.recent_block_timestamps.len();
```

Comment:
This having variance is not ideal. I remember talking about it and I see the problem of having to have a set amount of blocks before the current one in order to have the full 30 (or whatever the number) deltas, but validators already need those.
Since the canonical skip block will just have a fixed delta I think this should always be a fixed number.